Result/Result.swift

```diff
@@ -200,5 +200,11 @@ public func >>- <T, U, Error> (result: Result<T, Error>, @noescape transform: T
 	return result.flatMap(transform)
 }
 
+/// Returns a new Result by mapping `Success`es’ values using `transform`, or wrapping errors thrown as `Failure`s’.
+///
+/// This is a synonym for `tryMap`.
+public func >>- <T, U, Error> (result: Result<T, Error>, @noescape transform: T throws -> U) -> Result<U, Error> {
+    return result.tryMap(transform)
+}
```

Comment:
@robrix @neilpa  I agree it's good to introduce new operators sparingly. The newly proposed `map` behaves a lot like `flatMap`. The problem is that you can't currently mix and match `>>-` with `flatMap` or the newly proposed `map`. I wonder if this can be fixed by adjusting the precedence of `>>-`.